planner/core/find_best_task.go

```diff
@@ -469,6 +469,30 @@ func (ds *DataSource) findBestTask(prop *property.PhysicalProperty) (t task, err
 				p: dual,
 			}, nil
 		}
+		if !ds.isPartition && len(path.Ranges) > 0 {
```

Comment:
We can support Partition table with `len(path.Ranges) == 1`.

Reply:
Yes, you are right!